### ISSUE TITLE
Weird behaviour when reading windowed ranges

### ISSUE THREAD

**Issue by lmmx** [lmmx/range-streams#30]
```py
from range_streams import RangeStream, _EXAMPLE_URL
s = RangeStream(url=_EXAMPLE_URL, single_request=True)
s.add((0,3))
s.read(3)
```

gives the expected 3 bytes

but

```py
from range_streams import RangeStream, _EXAMPLE_URL
s = RangeStream(url=_EXAMPLE_URL, single_request=True)
s.add((0,3))
s.read(1)
s.read(1)
s.read(1)
s.read(1)
s.read(1)
s.read(1)
```

shows you can just keep reading the bytes, there’s no gating of the stream done at all.

- The active range response is not ‘decommissioned’ in the normal mode: like any stream, you can keep reading from it but you’ll just get the empty byte `b''`. This could be implemented by checking the `tell` on the response iterator and making it dependent on the byte range of the active range response (?)

**Comment by lmmx** [comment]
It may be necessary to ‘fake out’ the active range response after it’s been exhausted.

To act like an exhausted partial content request (when in single request mode), you could intercept the `active_range_response.read()` and `active_range_repsonse.tell()` calls via the RangeStream `read`/`tell` methods themselves, and where the `active_range_response` property would report its RangeResponse object with normal stream capabilities, when in single request mode it could actually create a faked RangeResponse with a dummy read method **only in the event that it has been consumed**.

There remains the problem of how to stop it reading too many bytes from the stream (and solving that may in fact make the previous suggestion unnecessary/overbaked)

**Comment by lmmx** [comment]
Maybe:
- set the buffer size
- set a default value on read in the case it's called without argument

But I don't think either of these will limit the actual stream's ability to return too much of the underlying stream

**Comment by lmmx** [comment]
Resolved, see #32
